File: 3550-smallest-index-with-digit-sum-equal-to-index/3550-smallest-index-with-digit-sum-equal-to-index.c
int digitsum(int x) {
    int summation = 0;

    while(x > 0) {
        int digit = x % 10;
        summation += digit;
        x = x / 10;
    }

    return summation;
}

int smallestIndex(int* nums, int numsSize) {

    for(int i = 0; i < numsSize; i++) {
        int addition = digitsum(nums[i]);

        if(addition == i) {
            return i;
        }
    }

    return -1;
}

File: 3550-smallest-index-with-digit-sum-equal-to-index/3550-smallest-index-with-digit-sum-equal-to-index.py
class Solution:

    def addition(self, x):
        summation = 0
        while x > 0:
            digit = x%10
            summation = summation+digit
            x = x//10
        return summation
    def smallestIndex(self, nums: List[int]) -> int:
        for i in range(len(nums)):
            summation = self.addition(nums[i])
            if summation == i:
                return i
                break
        return -1